lib/client.js

```diff
@@ -1199,6 +1203,334 @@ const PreClient = crypto => {
+     * @param {KeyPair} encryptionKeyPair Object that has encryption public and private keys
+     * @param {string} apiUrl Url of the TozStore api that you want to hit (Default is recommended).
+     *
+     * @returns {Note} A note from tozstore unencrypted with the client's keys.
+     */
+    static async readNote(
+      noteId,
+      signingKeyPair,
+      encryptionKeyPair,
+      apiUrl = DEFAULT_API_URL
+    ) {
+      let anonAuth = await AuthRequest.anonymousAuth(
+        crypto,
+        signingKeyPair.publicKey,
+        signingKeyPair.privateKey,
+        apiUrl
+      )
+      return Client.internalReadNote(anonAuth, encryptionKeyPair, noteId, apiUrl)
+    }
+
+    /**
+     * ReadNoteByName is a static method used to read a note by name,
+     * allowing you to supply your own signingKey and encryptionKey pairs.
+     *
+     * PLEASE NOTE: only notes written by a client, not the static writeNote method,
+     * can have a noteName attached.
+     *
+     * @param {string} noteName  name given to this note with premium features
+     * @param {KeyPair} signingKeyPair Object that has signing public and private keys
+     * @param {KeyPair} encryptionKeyPair Object that has encryption public and private keys
+     * @param {string} apiUrl Url of the TozStore api that you want to hit (Default is recommended).
+     *
+     * @returns {Note} A note from tozstore unencrypted with the client's keys.
+     */
+    static async readNoteByName(
+      noteName,
+      signingKeyPair,
+      encryptionKeyPair,
+      apiUrl = DEFAULT_API_URL
+    ) {
+      let anonAuth = await AuthRequest.anonymousAuth(
+        crypto,
+        signingKeyPair.publicKey,
+        signingKeyPair.privateKey,
+        apiUrl
+      )
+      return Client.internalReadNoteByName(anonAuth, encryptionKeyPair, noteName, apiUrl)
+    }
+
+    /**
+     * WriteNote is a static method that encrypts a note and sends it to TozStore,
+     * allowing you to supply your own signingKey and encryptionKey pairs.
+     *
+     * Using this method you are not allowed to provide premium options to TozStore,
+     * such as additional views, extended expiration time, etc.
+     *
+     * @param {string} type  Type of note to write
+     * @param {object} data  A hashmap of the data to encrypt and store
+     * @param {object} plainMeta A hashmap of unencrypted tags for data
+     * @param {object} fileMeta A hashmap of unencrypted tags for large files
+     * @param {string} recipientSigningKey signing key of the reader of this note
+     * @param {KeyPair} signingKeyPair Object that has signing public and private keys
+     * @param {string} apiUrl Url of the TozStore api that you want to hit (Default is recommended).     * @param {EncryptionKeyPair} encryptionKeyPair Object that has encryption public and private keys
+     *
+     * @returns {Note} A response from TozStore; the note that has been written.
+     */
+    static async writeNote(
+      type,
```

Comment:
in general type is significantly less important in notes than in records. I would re-order these parameters, possibly with an options object for writing notes -- type, plain, file meta are all optional items.